### PR TITLE
API-17: Get a list of categories

### PR DESCRIPTION
**Pull request by momoss**
Get a list of categories
https://akeneo.atlassian.net/browse/API-17

Call /api/rest/v1/categories
Response 200

```
{"items":[{"code":"master","parent":null,"labels":{"en_US":"Master catalog","de_DE":"Hauptkatalog","fr_FR":"Catalogue principal"}}]}
```

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | :negative_squared_cross_mark:
| Added Behats                      | :negative_squared_cross_mark:
| Added integration tests           | :white_check_mark:
| Changelog updated                 | :negative_squared_cross_mark:
| Review and 2 GTM                  | :white_check_mark:
| Micro Demo to the PO (Story only) | :negative_squared_cross_mark:
| Migration script                  | :negative_squared_cross_mark:
| Tech Doc                          | :negative_squared_cross_mark:

:white_check_mark: Done and pass

:red_circle: Done but fail

:clock1: Done but pending

:negative_squared_cross_mark: Not needed

### CODE REVIEW

**Comment by momoss** [comment]
Waitinf for https://github.com/akeneo/pim-community-dev/pull/5507

**Comment by momoss** [comment]
Need https://github.com/akeneo/pim-community-dev/pull/5507, before merge

File: src/Pim/Bundle/ApiBundle/Controller/Rest/CategoryController.php [hunk 1]
```diff
@@ -52,4 +52,18 @@ public function getAction(Request $request, $code)
...
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function listAction(Request $request)
+    {
+        $categories = $this->repository->findBy([], [], null, 0);
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/Controller/Rest/CategoryController.php]: please read the card API-17 to fix the order

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Category/ListCategoryIntegration.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace tests\integration\Pim\Bundle\ApiBundle\Controller\Rest\Category;
+
+use Symfony\Component\HttpFoundation\Response;
+use Test\Integration\TestCase;
+
+class ListCategoryIntegration extends TestCase
+{
+    public function testListCategories()
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Category/ListCategoryIntegration.php]: add a test to be sure nothing is returned if limit is bigger than the count of categories

**Pull request merged**